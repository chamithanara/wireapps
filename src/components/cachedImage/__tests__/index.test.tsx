import React from 'react';
import { render, screen, waitFor } from '@testing-library/react-native';

import FastImage from 'react-native-fast-image';
import CachedImage from '../index';

describe('Rendering Cached Image', () => {
    test('Snapshot of the initial render', async () => {
        render(<CachedImage uri="" />);
        await waitFor(() => expect(screen.toJSON()).toMatchSnapshot());
    });

    test('Snapshot with resizeMode contain prop', async () => {
        render(
            <CachedImage
                uri="https://unsplash.it/400/400?image=1"
                resizeMode={FastImage.resizeMode.contain}
                priority={FastImage.priority.normal}
            />
        );
        await waitFor(() => expect(screen.toJSON()).toMatchSnapshot());
    });

    test('Snapshot with resizeMode center prop', async () => {
        render(
            <CachedImage
                uri="https://unsplash.it/400/400?image=1"
                resizeMode={FastImage.resizeMode.center}
                priority={FastImage.priority.normal}
            />
        );
        await waitFor(() => expect(screen.toJSON()).toMatchSnapshot());
    });
});
