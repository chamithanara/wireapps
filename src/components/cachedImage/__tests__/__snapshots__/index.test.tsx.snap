// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Cached Image Snapshot of the initial render 1`] = `
<Image
  source={
    {
      "testUri": "../../../src/assets/images/logo/wireApps.png",
    }
  }
  style={
    [
      {},
      {
        "resizeMode": "cover",
      },
    ]
  }
/>
`;

exports[`Rendering Cached Image Snapshot with resizeMode center prop 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "column",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <ActivityIndicator
      color="#111111"
      size={25}
    />
  </View>,
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "zIndex": -1,
          },
          {},
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      onFastImageError={[Function]}
      onFastImageLoadEnd={[Function]}
      onFastImageLoadStart={[Function]}
      resizeMode="center"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>,
]
`;

exports[`Rendering Cached Image Snapshot with resizeMode contain prop 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "column",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <ActivityIndicator
      color="#111111"
      size={25}
    />
  </View>,
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "zIndex": -1,
          },
          {},
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      onFastImageError={[Function]}
      onFastImageLoadEnd={[Function]}
      onFastImageLoadStart={[Function]}
      resizeMode="contain"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>,
]
`;
